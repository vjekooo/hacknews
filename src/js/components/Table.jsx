
import React from 'react'
import PropTypes from 'prop-types'
import Button from './Button'

const Table = (props) => {
  const { result, onDismiss } = props
  return (
    <div className="table">
      {
        result
          .map(item =>
            <div key={item.objectID} className="table-row">
              <span>
                <a href={item.url}>{item.title}</a>
              </span>
              <span>{item.author}</span>
              <span>{item.num_comments}</span>
              <span>{item.points}</span>
              <Button
                className="button-inline"
                onDismiss={() => { onDismiss(item.objectID) }}
              >
                Dismiss
              </Button>
            </div>
          )
      }
    </div>
  )
}

Table.propTypes = {
  result: PropTypes.array.isRequired,
  onDismiss: PropTypes.func
}

export default Table
