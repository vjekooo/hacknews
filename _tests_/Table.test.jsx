
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'
import Table from '../src/js/components/Table'

describe('Table', () => {

  const props = {
    result: [
      { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
      { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
    ],
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    render(<Table { ...props } />, div)
  })
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Table { ...props } />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})