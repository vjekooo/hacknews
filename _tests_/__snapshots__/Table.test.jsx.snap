// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has a valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span>
      <a
        href={undefined}
      >
        1
      </a>
    </span>
    <span>
      1
    </span>
    <span>
      1
    </span>
    <span>
      2
    </span>
    <button
      className="button-inline"
      onClick={[Function]}
      type="button"
    >
      Dismiss
    </button>
  </div>
  <div
    className="table-row"
  >
    <span>
      <a
        href={undefined}
      >
        2
      </a>
    </span>
    <span>
      2
    </span>
    <span>
      1
    </span>
    <span>
      2
    </span>
    <button
      className="button-inline"
      onClick={[Function]}
      type="button"
    >
      Dismiss
    </button>
  </div>
</div>
`;
